protected.go

```diff
@@ -180,7 +190,7 @@ func (p *Protector) dial(op ops.Op, network, addr string, timeout time.Duration)
 	}
 	copy(conn.ip[:], ip.To4())
 
-	socketFd, err := syscall.Socket(syscall.AF_INET, syscall.SOCK_STREAM, 0)
+	socketFd, err := syscall.Socket(syscall.AF_INET, socketType, 0)
```

Comment:
LGTM, @oxtoacart! Do we also want to add support for deciding between syscall.AF_INET vs. syscall.AF_INET6 for the address family while we're in here too? This looks like a good reference: https://github.com/cofyc/cloudflare-go/blob/master/src/net/ipsock_posix.go#L108